tests/cpp/c_api/test_c_api.cc

```diff
@@ -63,3 +63,39 @@ TEST(c_api, XGDMatrixCreateFromMat_omp) {
     delete dmat;
   }
 }
+
+TEST(c_api, XGDMatrixCopyDataCSR) {
+  omp_set_num_threads(2);
```

Comment:
Store the original value first, and change it back at the end.  Don't assume it's running single threaded.  You can create a context class with constructor and destructor doing these jobs.